### ISSUE TITLE
Error deleting assertions via `datahub delete --entity-type assertions`

### ISSUE THREAD

**Issue by stevenayers** [datahub-project/datahub#9946]
**Describe the bug**
When trying to delete assertions, you receive a 500 error.

**To Reproduce**
1. Create some assertions
2. Run `datahub delete --entity-type assertion`

**Expected behavior**
Assertions are deleted

**Screenshots**
<img width="2352" alt="Screenshot 2024-02-28 at 05 31 44" src="https://github.com/datahub-project/datahub/assets/16361214/a235bdb6-aa36-4d55-9f70-d97b02da3d5e">

**Desktop (please complete the following information):**
 - CLI: 0.12.1.5
 - GMS: 0.11

**Additional context**
This could be related to the fact that the associated dataset was deleted previously using `datahub delete --env DEV`

Assertions should be tied to an environment too, so that when you delete an environment, you also delete the assertions associated with the datasets in that environment.

**Comment by stevenayers** [comment]
.

**Comment by hsheth2** [comment]
@stevenayers yup this is a gap - our graphql endpoints don't have a mechanism for listing assertions (since we usually see them as associated with datasets).

I believe urn-based deletes should still work - could you confirm that?
